site/en/blog/third-party-scripts/index.md

```diff
@@ -0,0 +1,264 @@
+
+1.  **Size and Script Execution Costs**: Third-parties often aim to provide significant functionality "just" by dropping in a script or iframe element into your page. These elements then pull in scripts and resources that are significant in size and take longer to download and execute. Too much JavaScript keeps the main thread busy longer, blocks rendering, and delays user interactions. Some of the top third-parties have been known to block the main thread from [42 ms to 1.6 s](https://almanac.httparchive.org/en/2021/third-parties#fig-14) for more than 50% of the sites analysed. A blocked main thread results in a high [Total Blocking Time (TBT)](https://web.dev/tbt/) which is one of the metrics that affects the performance score for the site. Additionally, it also delays response to user interactions and degrades the [Interaction to Next Paint (INP)](https://web.dev/inp/) metric used to measure the responsiveness of websites. Thus, script execution costs have a significant impact on performance.
+
+2.  **Number**: On average, websites use around [21 different third-parties](https://almanac.httparchive.org/en/2021/third-parties#fig-4) on mobile and web. Often, third-party tags are added by tag management tools that are not directly controlled by the technical/development teams. Tags which are not required may be added by other teams without a review process and never get removed. These can significantly impact user experience, page weight, or CPU utilisation. Establishing a [governance process](https://web.dev/controlling-third-party-scripts/) can address such situations and allow developers to audit each provider's impact. It would help if developers had ready data available for all third-parties that provide a specific function with their performance impact, benefits, and trade-offs for comparison. Another challenge teams face is that for many sites, their third-party tags only run in production but not in their development environments, making it more challenging for developers to test them.
+
+3.  **Network**: Since third-parties are hosted on different origins, browsers have to make a larger number of connections to download content from them. The different connections cannot coordinate download based on priority, resulting in network contention and suboptimal network use. This can further delay the page load if [the proper optimizations](https://web.dev/preconnect-and-dns-prefetch/#:~:text=Adding%20rel%3Dpreconnect%20to%20a,time%20the%20browser%20requests%20them.) are not considered.
```

Comment:
```suggestion
3.  **Network**: Since third-parties are hosted on different origins, browsers have to make a larger number of connections to download content from them. The different connections cannot coordinate download based on priority, resulting in network contention. This can further delay the page load if [the proper optimizations](https://web.dev/preconnect-and-dns-prefetch/#:~:text=Adding%20rel%3Dpreconnect%20to%20a,time%20the%20browser%20requests%20them.) are not considered.
```

Reply:
This is a total style nit, but I think we can cull a few words, as network contention is already a suboptimal outcome of slower performing architectures.